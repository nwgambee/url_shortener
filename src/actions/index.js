export const setUrls = (url) => ({
  type: 'SET_URLS',
  url
});
