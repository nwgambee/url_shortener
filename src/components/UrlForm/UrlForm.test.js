import { UrlForm, mapDispatchToProps } from '../UrlForm/UrlForm'
import { shallow } from 'enzyme';
import React from 'react';
import { setUrls } from '../../actions/index'



describe('UrlForm', () => {
  let wrapper;
  it('should match the snapshot', () => {
    wrapper = shallow(<UrlForm />)
    expect(wrapper).toMatchSnapshot();
  })
  describe('Method Tests', () => {
    it('should clear state when clearInputs is called', () => {
      wrapper = shallow(<UrlForm />)
      const initialState = {
        title: 'some title',
        urlToShorten: 'some url'
      };
      const expected = {
        title: '',
        urlToShorten: ''
      }
      wrapper.setState(initialState)
      wrapper.instance().clearInputs()
      expect(wrapper.state()).toEqual(expected)
    })
    it('should update state.title when handleNameChange is called', () => {
      wrapper = shallow(<UrlForm />)
      const initialState = {
        title: '',
        urlToShorten: ''
      };
      const expected = 'new shortened url'
      const mockEvent = {
        target: {name: 'title', value: 'new shortened url'}
      }
      wrapper.setState(initialState)
      wrapper.instance().handleNameChange(mockEvent)
      expect(wrapper.state('title')).toEqual(expected)
    })
    it('should update state.urlToShorten when handleNameChange is called', () => {
      wrapper = shallow(<UrlForm />)
      const initialState = {
        title: '',
        urlToShorten: ''
      };
      const expected = 'longerurl.com'
      const mockEvent = {
        target: {name: 'urlToShorten', value: 'longerurl.com'}
      }
      wrapper.setState(initialState)
      wrapper.instance().handleNameChange(mockEvent)
      expect(wrapper.state('urlToShorten')).toEqual(expected)
    })
  })

  describe('MapDispatchToProps tests', () => {
    it('should call dispatch with the setUrls action when setUrls is called', () => {
      const mockUrl = {longUrl: 'long url', shortUrl: 'shorter url'}
      const mockDispatch = jest.fn();
      const actionToDispatch = setUrls(mockUrl);
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.setUrls(mockUrl)
  
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  
  })
})