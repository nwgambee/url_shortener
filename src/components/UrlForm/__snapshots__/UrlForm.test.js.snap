// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UrlForm should match the snapshot 1`] = `
<form>
  <input
    name="title"
    onChange={[Function]}
    placeholder="Title..."
    type="text"
    value=""
  />
  <input
    name="urlToShorten"
    onChange={[Function]}
    placeholder="URL to Shorten..."
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
  >
    Shorten Please!
  </button>
</form>
`;
